runtime/manifest-parser.peg

```diff
@@ -321,11 +321,23 @@ ParticleArgumentDirection
   }
 
 ParticleArgumentType
-  = VariableType
+  = ConstraintType
+  / VariableType
   / SchemaInline
   / ReferenceType
   / ListType
 
+ConstraintType
+  = variable:VariableType whiteSpace 'with' whiteSpace type:ParticleArgumentType
+  {
+    return {
+      kind: 'constraint-type',
+      location: location,
```

Comment:
nit: this can just be `location`?

Reply:
It should have been `location: location(),` but this code no longer exists.